### ISSUE TITLE
Compilation error due using recursive inline 

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#41226]
**Godot version:**
3.2.3.rc.custom_build.23b553ba0

**OS/device including version:**
Github CI Ubuntu 20.04

**Issue description:**

Error
```
In file included from drivers/gles2/rasterizer_canvas_base_gles2.h:34,
                 from drivers/gles2/rasterizer_canvas_gles2.h:34,
                 from drivers/gles2/rasterizer_canvas_gles2.cpp:31:
drivers/gles2/rasterizer_array_gles2.h: In member function 'T* RasterizerArrayGLES2<T>::request_with_grow() [with T = RasterizerCanvasGLES2::BSortItem]':
drivers/gles2/rasterizer_array_gles2.h:107:20: error: inlining failed in call to always_inline 'T* RasterizerArrayGLES2<T>::request_with_grow() [with T = RasterizerCanvasGLES2::BSortItem]': recursive inlining
  107 |  _FORCE_INLINE_ T *request_with_grow() {
      |                    ^~~~~~~~~~~~~~~~~
drivers/gles2/rasterizer_array_gles2.h:111:29: note: called from here
  111 |    return request_with_grow();
      |                             ^
scons: *** [drivers/gles2/rasterizer_canvas_gles2.x11.opt.64s.o] Error 1
```

probably removing _FORCE_INLINE_ from
https://github.com/godotengine/godot/blob/abf019d810e7249a50b2d9b6b99aa217aa50a4d2/drivers/gles2/rasterizer_array_gles2.h#L107
will fix this 

Failed job -https://github.com/godotengine/godot/runs/980316761?check_suite_focus=true

**Steps to reproduce:**
Compile godot with this flags
```
scons -j10 verbose=yes warnings=all werror=yes platform=x11 tools=no target=release debug_symbols=full use_ubsan=yes use_asan=yes
```

**Comment by lawnjelly** [comment]
Probably me, however I don't like using FORCED_INLINE for this and other reasons. I think one of the reviewers got me to change it. :smiling_imp:

I don't even tend to use inline keyword as it is usually ignored and compiler will do where appropriate. I'll do a PR to change them.

**Comment by akien-mga** [comment]
Fixed by #41231.
